File: 04_quick_sort/JS/05_quick_sort.js
const quickSort = (array) => {
    if (array.length < 2) {
        return array;
    }
    const pivot = array[0];
    const less = array.slice(1).filter(key => key <= pivot);
    const greater = array.slice(1).filter(key => key > pivot);
    return [...quickSort(less), pivot, ...quickSort(greater)]
};

console.log(quickSort([10, 5, 2, 3])); // [2, 3, 5, 10]

File: 04_quick_sort/Python/05_quick_sort.py
def quick_sort(array):
    if len(array) < 2:
        return array # Базовый случай: массивы с 0 и 1 элементом уже "отсортированы"
    else: # Рекурсивный случай
        pivot = array[0] # Опорный элемент
        less = [i for i in array[1:] if i <= pivot] # подмассив всех элементов меньших опорного
        greater = [i for i in array[1:] if i > pivot] # подмассив всех элементов больших опорного
        return quick_sort(less) + [pivot] + quick_sort(greater)

print(quick_sort([10, 5, 2, 3, 0, 15, -1])) 